cmd/bosun/expr/funcs.go

```diff
@@ -657,7 +673,11 @@ func timeGraphiteRequest(e *State, T miniprofiler.Timer, req *graphite.Request)
 		}
 		var val interface{}
 		val, err = e.cache.Get(key, getFn)
-		resp = val.(graphite.Response)
+		// hand out copy of the value, so callers can't accidentally modify data in cache
+		if val, ok := val.(*graphite.Response); ok && val != nil {
```

Comment:
Yes, but that doesn't happen with graphite data because it is entirely copied later on.